### PR TITLE
Daniel Ogbuti-Building User Interfaces for Celo

### PR DESCRIPTION
**Pull request by dahnny**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+title: "Building Dynamic and Scalable User Interfaces for Celo ERC721 Smart Contracts using React: A Comprehensive Introduction"
+description: Integrating your smart contract with a frontend framework like React can provide a seamless and user-friendly experience for interacting with your contract
+authors:
+  - name: Daniel Ogbuti
+    title: Web3 Developer,
+    url: https://github.com/dahnny
+    image_url: https://github.com/dahnny.png
+tags: ["celo", "intermediate", "celosage", "react"]
+hide_table_of_contents: true
+slug: /tutorials/building-dynamic-user-interfaces-for-erc721
```
Comment (yafiabiyyu) [inline blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md:11]:
```suggestion
slug: /tutorials/building-dynamic-and-scalable-user-interfaces-for-celo-erc721-smart-contracts-using-react'
```

Comment (yafiabiyyu) [reply blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md:11]: For the "slug", my suggestion is to use a name such as "building-dynamic-and-scalable-user-interfaces-for-celo-erc721-smart-contracts-using-react". This name will better represent the title of your tutorial. Also, remember to replace the section contained in users.tsx.

File: blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+  }
+};
+
+```
+This  function fetches all the NFTs (non-fungible tokens) available on a marketplace by calling methods on the minter and market contracts. 
+
+It does this by first getting the length of the listings array, then looping through each listing and creating a Promise for each one. 
+Inside the Promise, it retrieves the URI of the token and uses it to fetch metadata for the token. It also gets the owner of the token and other relevant data from the listing. 
+
+It then returns an array of all the Promises, which are resolved using Promise.all() to return an array of all the NFTs with their associated metadata. If there are any errors during this process, they are logged to the console.
```
Comment (yafiabiyyu) [inline blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md:411]:
```suggestion
It then returns an array of all the Promises, which are resolved using `Promise.all()` to return an array of all the NFTs with their associated metadata. If there are any errors during this process, they are logged to the console.
```

File: blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+      getAssets();
+    } catch (error) {
+      console.log({ error });
+      toast(<NotificationError text="Failed to create an NFT." />);
+    } finally {
+      setLoading(false);
+    }
+  };
+```
+It takes in several parameters such as the minterContract and marketContract, performActions, index, and tokenId. When executed, the function sets the loading state to true, buys the NFT using the buyNft function, and then displays a success notification.
```
Comment (yafiabiyyu) [inline blog/05-09-23-building-dynamic-user-interfaces-for-erc721/building-dynamic-user-interfaces-for-erc721.md:539]:
```suggestion
It takes in several parameters such as the `minterContract` and `marketContract`, `performActions`, `index`, and `tokenId`. When executed, the function sets the loading state to true, buys the NFT using the `buyNft` function, and then displays a success notification. 
```

File: src/data-tutorials/users.tsx [hunk 4]
```diff
@@ -2981,6 +2981,13 @@ const Users: User[] = [
     website: 'blog/tutorials/how-to-create-a-decentralized-application-using-react-celo',
     tags: ['celosage', 'smartcontract', 'solidity', 'react', 'hardhat', 'crowdfunding', 'intermediate'],
   },
+  {
+    title: 'Building Dynamic and Scalable User Interfaces for Celo ERC721 Smart Contracts using React- A Comprehensive Introduction',
+    description: 'Integrating your smart contract with a frontend framework like React can provide a seamless and user-friendly experience for interacting with your contract',
+    preview: require('./showcase/intermediate/building-dynamic-user-interfaces-for-erc721.png'),
+    website: 'blog/tutorials/building-dynamic-user-interfaces-for-erc721',
```
Comment (yafiabiyyu) [inline src/data-tutorials/users.tsx:2988]:
```suggestion
    website: 'blog/tutorials/building-dynamic-and-scalable-user-interfaces-for-celo-erc721-smart-contracts-using-react',
```

**Review by yafiabiyyu — COMMENTED** [review]
### Regarding the review results, please take note of the following recommendations:

1. Rename the slug to reflect the title of your tutorial.
2. Consider moving the code snippets for MyNFT.sol and MyNFTMarket.sol to the "Getting Started" section to ensure a more logical sequence of tutorial content.
3. Make a few minor revisions to the code in the markdown files.

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**